crates/goose/src/providers/anthropic.rs

```diff
@@ -57,25 +60,38 @@ impl AnthropicProvider {
         })
     }
 
-    async fn post(&self, payload: Value) -> Result<Value, ProviderError> {
+    async fn post(&self, headers: HeaderMap, payload: Value) -> Result<Value, ProviderError> {
         let base_url = url::Url::parse(&self.host)
             .map_err(|e| ProviderError::RequestFailed(format!("Invalid base URL: {e}")))?;
         let url = base_url.join("v1/messages").map_err(|e| {
             ProviderError::RequestFailed(format!("Failed to construct endpoint URL: {e}"))
         })?;
 
+        if std::env::var("GOOSE_DEBUG").is_ok() {
+            println!(
+                "\nRequest:\n{}\n",
+                serde_json::to_string_pretty(&payload).unwrap()
+            );
+        }
+
         let response = self
             .client
             .post(url)
-            .header("x-api-key", &self.api_key)
-            .header("anthropic-version", "2023-06-01")
+            .headers(headers)
             .json(&payload)
             .send()
             .await?;
 
         let status = response.status();
         let payload: Option<Value> = response.json().await.ok();
 
+        if std::env::var("GOOSE_DEBUG").is_ok() {
```

Comment:
nit: do we use this standard elsewhere?  definitely seems useful but i'm not sure it makes more sense necessarily than `tracing`